x-pack/filebeat/tests/integration/managerV2_test.go

```diff
@@ -504,9 +504,17 @@ func TestRecoverFromInvalidOutputConfiguration(t *testing.T) {
 	// We use `success` to signal the test has ended successfully
 	// if `success` is never closed, then the test will fail with a timeout.
 	success := make(chan struct{})
+
+	// succeededChannelClosed is set to true whenever the `success`
+	// channel is closed. The Filestream input now is reporting its state
+	// to the Elastic-Agent, which cases more checkins to happen, thus the
+	// `success` channel was being close twice. `succeededChannelClosed`
+	// prevents that from happening.
+	succeededChannelClosed := atomic.Bool{}
 	// The test is successful when we reach the last element of `protoUnits`
 	onObserved := func(observed *proto.CheckinObserved, protoUnitsIdx int) {
-		if protoUnitsIdx == len(protos)-1 {
+		if protoUnitsIdx == len(protos)-1 && !succeededChannelClosed.Load() {
```

Comment:
I'm not sure this guarantees that close will only get called once.  Is it possible for the condition to be evaluated, then go routine put to sleep and another go routine run the same code and call close?  In that case we would still call close twice.

Could we just use the `sync.Once` pattern to make sure that close is only called once?

Reply:
You're right Lee, I'll make this improvement PR.

Reply:
Here is the PR with the fix: https://github.com/elastic/beats/pull/40199